profiles/management/commands/gen_realistic_users.py

```diff
@@ -0,0 +1,254 @@
+        if hasattr(model_obj, k) and not is_relation_field(model_obj, k):
+            setattr(model_obj, k, v)
+    if save:
+        model_obj.save()
+    return model_obj
+
+
+# User data deserialization
+
+class CachedModelDeserializer:
+    """
+    Base class for deserializing a model used for edX data caching (eg: Enrollments, Certificates)
+    """
+    model_cls = None
+
+    @classmethod
+    def _get_edx_course_key(cls, data):   # pylint: disable=unused-argument
+        """Gets the edx_course_key value from the data dictionary"""
+        return ''
+
+    @classmethod
+    def _fill_in_missing_data(cls, data, user, course_run):   # pylint: disable=unused-argument
+        """Fills in data using the associated User and CourseRun objects"""
+        return data
+
+    @classmethod
+    def deserialize(cls, user, data, course_runs):
+        """
+        Creates a new cached model object and returns the associated CourseRun
+        """
+        edx_course_key = cls._get_edx_course_key(data)
+        course_run = first_matching_item(course_runs, lambda cr: cr.edx_course_key == edx_course_key)
+        data = cls._fill_in_missing_data(data, user, course_run)
+        cls.model_cls.objects.create(
+            user=user,
+            course_run=course_run,
+            data=data,
+            last_request=CACHED_MODEL_LAST_REQUEST
+        )
+        return course_run
+
+
+class CertificateDeserializer(CachedModelDeserializer):
+    """
+    CachedCertificate deserializer
+    """
+
+    model_cls = CachedCertificate
+    data_key = 'certificates'
+
+    @classmethod
+    def _get_edx_course_key(cls, data):
+        return data['course_id']
+
+    @classmethod
+    def _fill_in_missing_data(cls, data, user, course_run):
+        data['username'] = user.username
+        return data
+
+
+class EnrollmentDeserializer(CachedModelDeserializer):
+    """
+    CachedEnrollment deserializer
+    """
+
+    model_cls = CachedEnrollment
+    data_key = 'enrollments'
+
+    @classmethod
+    def _get_edx_course_key(cls, data):
+        return data['course_details']['course_id']
+
+    @classmethod
+    def _fill_in_missing_data(cls, data, user, course_run):
+        data['user'] = user.username
+        data['course_details'].update({
+            'course_start': course_run.start_date.isoformat(),
+            'course_end': course_run.end_date.isoformat(),
+            'enrollment_start': course_run.enrollment_start.isoformat(),
+            'enrollment_end': course_run.enrollment_end.isoformat()
+        })
+        return data
+
+
+CACHED_MODEL_DESERIALIZERS = [EnrollmentDeserializer, CertificateDeserializer]
+
+
+def deserialize_user_data(user_data, course_runs):
+    """
+    Deserializes a dict of mixed User/Profile data and returns the newly-inserted User
+    """
+    user = User.objects.create(
+        email=user_data['email'],
+        first_name=user_data['first_name'],
+        last_name=user_data['last_name'],
```

Comment:
i think the reason i didn't do it here was b/c `deserialize_model_data` wants a model _object_ rather than a class. at the time, i didn't see a clean way to pass in a model class and figure out all the attributes that could be filled in, and i think User was complaining about required fields when i tried to instantiate an object then fill in attributes. i just took another look in the shell and discovered `<ModelClass>._meta.get_fields()`. might be able to leverage that. i'll take a look